pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1528,8 +1520,8 @@ private void makeStreamMetadataProvider(String reason) {
     if (_streamMetadataProvider != null) {
       closeStreamMetadataProvider();
     }
-    _segmentLogger.info("Creating new stream metadata provider, reason: {}", reason);
-    _streamMetadataProvider = _streamConsumerFactory.createStreamMetadataProvider(_clientId);
+    _segmentLogger.info("Creating new partition metadata provider, reason: {}", reason);
+    _streamMetadataProvider = _streamConsumerFactory.createPartitionMetadataProvider(_clientId, _partitionGroupId);
```

Comment:
Should we consider renaming the caller method name?

Reply:
I don't know why an interface called `StreamMetadataProvider` contains a method for both `createStreamMetadataProvider` (stream-level) and `createPartitionMetadataProvider` (partition-level). I am sure there some history to do this. 
what is your proposal here?

Reply:
> I don't know why an interface called StreamMetadataProvider contains a method for both createStreamMetadataProvider (stream-level) and createPartitionMetadataProvider (partition-level).

You mean `StreamConsumerFactory` right?

Just taking a closer look, this class is called `HLRealtimeSegmentDataManager`, which is a high level data manager for realtime segment. So it should use the stream-level instead of the partition-level provider. @navina We should revert this change here.

Reply:
Ok. I renamed the method to `createPartitionMetadataProvider`

Reply:
Sorry for the late comment here. Somehow I forgot to submit the review yesterday. 

By taking a closer look, the purpose of `_streamMetadataProvider` is to fetch the latest value of partitionCount from the stream provider, which shouldn't associate with any particular partitionId here. We should stick to using the stream-level stream metadata provider here.